packages/blade/src/components/DatePicker/DateInput.web.tsx

```diff
@@ -3,40 +3,160 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import React from 'react';
 import { useDatesContext } from '@mantine/dates';
-import type { DatePickerInputProps } from './types';
-import { getFormattedDate } from './utils';
+import type { DatePickerInputProps, DateInputProps } from './types';
+import {
+  getFormattedDate,
+  rangeFormattedValue,
+  rangeInputPlaceHolder,
+  finalInputFormat,
+  getTextInputFormat,
+  validateAndParseDateInput,
+} from './utils';
+import { usePresetContext } from './QuickSelection/PresetContext';
 import BaseBox from '~components/Box/BaseBox';
-import { ArrowRightIcon, CalendarIcon } from '~components/Icons';
-import type { BaseInputProps } from '~components/Input/BaseInput';
-import { BaseInput } from '~components/Input/BaseInput';
-import { size as sizeTokens } from '~tokens/global';
-import { isReactNative, makeSize } from '~utils';
+import { TextInput } from '~components/Input/TextInput';
+import { isReactNative } from '~utils';
 import type { BladeElementRef, DataAnalyticsAttribute } from '~utils/types';
-import { useIsMobile } from '~utils/useIsMobile';
+import { CalendarIcon } from '~components/Icons';
 import { makeAnalyticsAttribute } from '~utils/makeAnalyticsAttribute';
 
+/**
+ * CRITICAL BEHAVIOR CASES - Verify when making changes:
+ *
+ * 1. VALUE PROP: Already formatted by getFormattedDate() - avoid double formatting
+ * 2. USER TYPING: Should format input AND sync with calendar selection
+ * 3. CALENDAR SELECTION: Should apply formatted value to input without re-formatting
+ * 4. SUBMIT (no footer): Blur/Enter should select current value
+ */
 const _DateInput = (
-  props: BaseInputProps,
+  props: DateInputProps,
   ref: React.ForwardedRef<BladeElementRef>,
 ): React.ReactElement => {
+  const { format, date, setControlledValue, leadingDropdown, tags, id, ...textInputProps } = props;
+  const [inputValue, setInputValue] = React.useState(['']);
+  const [validationError, setValidationError] = React.useState<string | undefined>(undefined);
+
+  const presetContext = usePresetContext();
+
+  // Determine selection type: prefer preset context calculation over props
+  // This handles "Today" presets that should display as single even though data is range
+  const isRange =
+    presetContext?.effectiveSelectionType === 'single'
+      ? false
+      : presetContext?.effectiveSelectionType === 'range' || props.selectionType === 'range';
+
+  // Remove slashes from formatted dates for internal input state management
+  const stripDelimiters = (str?: string): string => str?.replace(/\//g, '') ?? '';
+
+  // Sync internal input state with external formatted values from parent component
```

Comment:
Didn't understood what's happening can you add an example on whats happening? Also what is `textInputProps.value`?